File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        fp = 0
        sp = 1
        count = 0
        while sp < len(nums):
            if nums[fp] != nums[sp]:
                count += 1
                nums[fp + 1] = nums[sp]
                fp += 1
                sp += 1
            else:
                sp += 1
                

        return count + 1
        

File: Data Structures & Algorithms/move-zeroes/submission-0.java
class Solution {
    public void moveZeroes(int[] nums) {
        int len = nums.length;
        int start = 0;
        int zero = 0;

        while (start < len) {
            // find first non zero
            while (start < len && nums[start] == 0) start++;

            if (start < len) {
                nums[zero] = nums[start];
                zero++;
            }
            start++;
        }

        for (int i = zero; i < len; i++) {
            nums[i] = 0;
        }  
    }
}